Extensions/xkit_patches.js

```diff
@@ -915,6 +915,195 @@ XKit.extensions.xkit_patches = new Object({
 						tags: post.tags.join(","),
 					};
 				},
+				/**
+				 * Get the posts on the screen without the given tag
+				 * @param {String} without_tag - Class that the posts should not have
+				 * @param {Boolean} can_edit - Whether the posts must be editable
+				 * @return {Array<Object>} The posts
+				 */
+				get_posts: async function(without_tag, can_edit) {
+					var posts = [];
+
+					var selector = "[data-id]";
+
+					var editLabel = await XKit.interface.translate("Edit");
+
+					var selection = $(selector);
+					selection = selection.not("." + without_tag);
+
+					selection.each(function() {
+						var $this = $(this);
+
+						// If can_edit is requested and we don't have an edit post control,
+						// don't push the post
+
+						if (can_edit && $this.find(`[aria-label="${editLabel}"]`).length === 0) {
+							return;
+						}
+						posts.push($this);
+					});
+					return posts;
```

Comment:
this could be cleaned up in a few ways

1. `selector` doesn't *really* need to be its own variable since it's static and is only used once
2. using `[data-id]:not(.${without_tag})` should be more efficient, since the browser should be able to optimise the query
3. `editLabel` only needs to be set if `can_edit` is requested, and by extension,
4. the loop only needs to be run if `can_edit` is requested... and on the same note,
5. the loop would probably be better as a `.filter()`, since then we can keep it as a jQuery set more easily

```suggestion
					var $posts = $(`[data-id]:not(.${without_tag})`);

					if (can_edit) {
						const editLabel = await XKit.interface.translate("Edit");
						$posts = $posts.filter(function() {
							return $(this).find(`[aria-label="${editLabel}"]`).length !== 0;
						});
					}

					return $posts;
```

hopefully this code works, but i can't test it since i'm not at my desktop - if it's flawed, please correct it! and if i'm wrong about jQuery sets being iterable, you may need to adjust Quick Tags' code to use `.each()` instead